include/rdmnet/cpp/controller.h

```diff
@@ -0,0 +1,164 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************
+ * This file is a part of RDMnet. For more information, go to:
+ * https://github.com/ETCLabs/RDMnet
+ *****************************************************************************/
+
+/// \file rdmnet/cpp/controller.h
+/// \brief C++ wrapper for the RDMnet Controller API
+
+#ifndef RDMNET_CPP_CONTROLLER_H_
+#define RDMNET_CPP_CONTROLLER_H_
+
+#include <string>
+#include <vector>
+#include "etcpal/cpp/error.h"
+#include "etcpal/cpp/inet.h"
+#include "etcpal/cpp/uuid.h"
+#include "etcpal/log.h"
+#include "rdmnet/controller.h"
+
+namespace rdmnet
+{
+/// \defgroup rdmnet_controller_cpp Controller C++ Language API
+/// \ingroup rdmnet_controller
+/// \brief C++ Language version of the Controller API
+/// @{
+
+using ControllerHandle = rdmnet_controller_t;
+using ScopeHandle = rdmnet_client_scope_t;
+
+/// @}
+
+/// \brief A base class for a class that receives notification callbacks from a controller.
+/// \ingroup rdmnet_controller_cpp
+class ControllerNotify
```

Comment:
Find better naming for this interface. Name should be noun (e.g. ControllerNotifyHandler or something) and function names should be verbs (e.g. starting with Handle...)